cg/apps/lims/orderform.py

```diff
@@ -38,22 +39,24 @@ def check_orderform_version(document_title):
 
 def parse_orderform(excel_path: str) -> dict:
     """Parse out information from an order form."""
-    workbook = xlrd.open_workbook(excel_path)
+
+    workbook: Workbook = openpyxl.load_workbook(filename=excel_path, read_only=True, data_only=True)
 
     sheet_name = None
-    sheet_names = workbook.sheet_names()
+    sheet_names: List[str] = workbook.sheetnames
     for name in ["Orderform", "orderform", "order form"]:
         if name in sheet_names:
             sheet_name = name
             break
     if sheet_name is None:
         raise OrderFormError("'orderform' sheet not found in Excel file")
-    orderform_sheet = workbook.sheet_by_name(sheet_name)
 
-    document_title = get_document_title(workbook, orderform_sheet)
+    orderform_sheet: Worksheet = workbook[sheet_name]
+    document_title: str = get_document_title(workbook, orderform_sheet)
     check_orderform_version(document_title)
 
     raw_samples = relevant_rows(orderform_sheet)
+
```

Comment:
take a look at the codefactor warning! i think it can be flattened + refactored to return data if customer_ids == 1 and raise error otherwise

Reply:
I'm working on that in another PR!